src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -732,6 +771,41 @@ impl Drop for CanceledWorkunitGuard {
   }
 }
 
+#[derive(Clone)]
+struct MetricsData {
+  counters: Arc<Mutex<HashMap<SpanId, HashMap<Metric, i64>>>>,
```

Comment:
Perhaps mention why this is signed, as future readers might not see the design doc.